### ISSUE TITLE
Question about Sequential Reco training

### ISSUE THREAD

**Issue by berkecanrizai-getir** [massquantity/LibRecommender#365]
Hi, I am looking to train a sequential model using user + product features and sessions such as (1, 2, 3) -> 4 where user had few items in basket and I want to predict the next possible item. I checked the [seq_example](https://github.com/massquantity/LibRecommender/blob/64cb133d978abda68907d21194a154eb8b357b7a/examples/seq_example.py) file however, it's not clear to me how model is given the session info. 

In [data_info](https://github.com/massquantity/LibRecommender/blob/64cb133d978abda68907d21194a154eb8b357b7a/docs/source/internal/data_info.rst#L14) it says that interactions are time sorted however, from my understanding, interactions are not explicitly divided into separate sessions. So, how to train the model with such sessions? 

I have also seen the following issue https://github.com/massquantity/LibRecommender/issues/193

**Comment by massquantity** [comment]
In essence, there is no "session" concept in this library. All the interactions from a user are merged into a single sequence and processed sequentially during training. 

Sessions exhibit a broad range of lengths, spanning from just 1 to potentially over 10,000 interactions. However, neural networks require fixed input lengths, making it challenging to predetermine session lengths. If session management is of great importance to you, this library isn't equipped to fulfill that requirement.

Nonetheless, if you want to use the library, the sessions need to be "flattened" first. For example, say you have two sessions: (1, 2, 3) -> 4, (6, 7) -> 8 and a user `a`. In this scenario, the initial dataset ought to be transformed into the following format:
a  1
a  2
a  3
a  4
a  6
a  7
a  8

The training data, once transformed within the library, will manifest as follows:
x x x x 1
x x x 1 2
x x 1 2 3
x 1 2 3 4
1 2 3 4 6
2 3 4 6 7
3 4 6 7 8
where `x` represents the padding value, and we assume the predefined input length is 5.
